tests/components/vera/common.py

```diff
@@ -0,0 +1,839 @@
+            elif service_id == "urn:micasaverde-com:serviceId:SecuritySensor1":
+                if action == "SetArmed":
+                    status_variable_name = "armed"
+            elif service_id == "urn:upnp-org:serviceId:WindowCovering1":
+                if action == "SetLoadLevelTarget":
+                    status_variable_name = "level"
+            elif service_id == "urn:micasaverde-com:serviceId:DoorLock1":
+                if action == "NewTarget":
+                    status_variable_name = "locked"
+            elif service_id == "urn:upnp-org:serviceId:HVAC_UserOperatingMode1":
+                if action == "SetModeTarget":
+                    status_variable_name = "mode"
+            elif service_id == "urn:upnp-org:serviceId:HVAC_FanOperatingMode1":
+                if action == "SetMode":
+                    status_variable_name = "fanmode"
+            elif service_id == "urn:upnp-org:serviceId:TemperatureSetpoint1_Cool":
+                pass
+            elif service_id == "urn:upnp-org:serviceId:TemperatureSetpoint1_Heat":
+                pass
+            elif service_id == "urn:upnp-org:serviceId:TemperatureSetpoint1":
+                if action == "SetCurrentSetpoint":
+                    status_variable_name = "setpoint"
+            elif service_id == "urn:micasaverde-com:serviceId:Color1":
+                if action == "SetColorRGB":
+                    status_variable_name = "CurrentColor"
+
+            device = get_device(device_id, component_data)
+            status = get_device_status(device_id, component_data)
+
+            # Update the device and status objects.
+            if status_variable_name is not None:
+                device[status_variable_name] = state_variable_value
+                status[status_variable_name] = state_variable_value
+
+            # Update the state object.
+            status["states"] = [
+                state
+                for state in status.get("states", [])
+                if state.get("service") != service_id
+                or state.get("variable") != state_variable_name
+            ]
+            status["states"].append(
+                {
+                    "service": service_id,
+                    "variable": state_variable_name,
+                    "value": state_variable_value,
+                }
+            )
+
+            return ResponseStub(json={})
+
+        return ResponseStub(json={})
+
+    def init_controller(base_url: str) -> list:
+        nonlocal data_request
+        controller = VeraController(base_url)
+        controller.data_request = MagicMock(side_effect=data_request)
+        controller.start()
+        return [controller, True]
+
+    component_data = ComponentData(
+        sdata=deepcopy(response_sdata),
+        status=deepcopy(response_status),
+        lu_sdata=deepcopy(respone_lu_sdata),
+    )
+
+    patch("pyvera.init_controller", side_effect=init_controller).start()
```

Comment:
This patch is never stopped.

Reply:
Where do we stop this patch?

Reply:
Thanks for catching that again. Not sure why I resolved it earlier. At any rate, it's fixed now.